website/content/en/docs/advanced-topics/custom-bundle-validation.md

```diff
@@ -0,0 +1,124 @@
+
+```sh
+operator-sdk bundle validate --alpha-select-external path/validator1:path/validator2
+```
+
+## Writing a Custom Validator
+
+For a validator to work with `operator-sdk bundle validate` each of the files must:
+1. Be executable with appropriate permissions
+1. Return JSON to STDOUT in the [ManifestResult][manifest_result] format.
+
+### Custom Validator from Scratch
+
+Using the `errors` package from `operator-framework/api`, we can start
+by validating the correct number of arguments and marshaling a
+`ManifestResult` into STDOUT.
+
+```go
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+
+	"github.com/operator-framework/api/pkg/validation/errors"
+)
+
+func main() {
+
+	// we expect a single argument which is the bundle root.
+	// usage: validator-poc <bundle root>
+	if len(os.Args) < 2 {
+		fmt.Printf("usage: %s <bundle root>\n", os.Args[0])
+		os.Exit(1)
+	}
+
+	var validatorErrors []errors.Error
+	var validatorWarnings []errors.Error
+	result := errors.ManifestResult{
+		Name:     "Always Green Example",
+		Errors:   validatorErrors,
+		Warnings: validatorWarnings,
+	}
+	prettyJSON, err := json.MarshalIndent(result, "", "    ")
+	if err != nil {
+		fmt.Println("Invalid json")
+		os.Exit(1)
+	}
+	fmt.Printf("%s\n", string(prettyJSON))
+}
+```
+
+These custom validators can be written in any language. When executed on
+its own, this validator prints a json representation of ManifestResult.
+
+```sh
+go build main.go && ./main
+
+{
+    "Name": "Always Green Example",
+    "Errors": null,
+    "Warnings": null
+}
+```
+
+As long as the STDOUT follows the ManifestResult spec, it can be used
+with `operator-sdk bundle validate` command.
+
+```sh
+go build main.go && operator-sdk bundle validate ./bundle --alpha-select-external ./main
+
+INFO[0000] All validation tests have completed successfully 
+```
+
+From here, custom validator authors can read in the bundle and make any
+assertions necessary.
+
+Errors and Warnings are both implementations of the `error` interface
+and need `ErrorType`, `Level`, `Field`, `BadValue`, and `Detail`, which
+are all initialized by arbitrary strings.
+
+```go
+validatorErrors = []errors.Error{errors.Error{"someErrorType", "somelevel", "somefield", "somebadvalue", "somedetail"}}
+validatorWarnings = []errors.Error{errors.Error{"someWarningType", "somelevel", "somefield", "somebadvalue", "somedetail"}}
+```
+
+We can now rebuild and run the validator, which now shows errors. 
+```sh
+go build main.go && operator-sdk bundle validate ./bundle --alpha-select-external ./main
+
+WARN[0000] somelevel: Field somefield, Value somebadvalue: somedetail 
+ERRO[0000] somelevel: Field somefield, Value somebadvalue: somedetail 
+```
+
+### Composing Validators
+
+For users wishing to validate using `operator-framework/api` of a
+different version than what is built into the `operator-sdk` binary, it
+is possible to import the `validation` package from the `api` repository
+at an arbirary version. 
```

Comment:
```suggestion
You can also validate using a different version of the `operator-framework/api` 
than what is built into the `operator-sdk` binary. You can 
import the `validation` package from the `api` repository
at an arbitrary version. 
```
Typo correction and a suggestion to break up the long sentence.

Reply:
Would it help the user to have direct links to the repos and packages in this sentence?